File: 1/php/Row.php
<?php
class Row
{
    public $x,$y,$birthTime, $time,$r, $booling="notValid",$exTime=0;
    function __construct($x,$y,$r,$birthTime, $time)
    {
        $this->x=$x;
        $this->y=$y;
        $this->r=$r;
        $this->birthTime=$birthTime;
        $this->time=$time;
    }
    function toString(){
        return '<tr><td>' . $this->booling . '</td><td>' . sprintf('%.2e', $this->exTime) . '</td><td>' . $this->time . '</td>'
        . '<td>' . $this->x . '</td>'
        . '<td>' . $this->y . '</td>'
        . '<td>' . $this->r . '</td></tr>';
    }
    function isHit(){
        $this->booling="no";
        if($this->x>=0 && $this->x<=$this->r/2&& $this->y<=$this->r && $this->y>=0){ $this->booling="yes";}
        if($this->x>0 && $this->y<=0&&$$this->x**2+$this->y**2<=($this->r**2)/4) {$this->booling="yes";}
        if($this->x<0 && $this->y>=-$this->x-$this->r&&$this->y<=0) {$this->booling="yes";}
        $time_end = microtime(true);
        $this-> exTime=$time_end-$this->birthTime;

    }
   
}

?>

File: web2/src/main/java/model/Row.java
package model;

import java.io.Serializable;
import java.time.LocalTime;
import java.time.ZoneId;
import java.time.temporal.ChronoUnit;

public class Row implements Serializable {
    private double x;
    private double y;
    public Row(){
        this.exTime=System.nanoTime();
    }
    public Row(double x, double y, double r, ZoneId zone){
        this.x=x;
        this.y=y;
        this.r = r;
        this.birthTime=LocalTime.now(zone).truncatedTo(ChronoUnit.SECONDS);
        this.exTime=System.nanoTime();
    }

    public void isHit() {
        if (x>=0 &&y>=0 && y<=r && x<=r/2) booling = true;
        if (x>0 && y<=0 && x*x+y*y<=r*r/4) booling = true;
        if (x<0 && y<=0 && y>=-x-r) booling = true;
        exTime = -exTime + System.nanoTime();

    }
    @Override
    public String toString() {
        return ("<tr><td>" + booling + "</td><td>" + exTime + "</td><td>" + birthTime +  "</td>"
                + "<td name=xT>" + Math.round(x*1000.0)/1000.0 + "</td>"
                + "<td name=yT>" + Math.round(y*1000.0)/1000.0 + "</td>"
                + "<td name=rT>" + r +  "</td></tr>");
    }


    public double getX() {
        return x;
    }

    public void setX(double x) {
        this.x = x;
    }

    public double getY() {
        return y;
    }

    public void setY(double y) {
        this.y = y;
    }

    public double getR() {
        return r;
    }

    public void setR(double r) {
        this.r = r;
    }

    public LocalTime getBirthTime() {
        return birthTime;
    }

    public void setBirthTime(LocalTime birthTime) {
        this.birthTime = birthTime;
    }

    public boolean isBooling() {
        return booling;
    }

    public void setBooling(boolean booling) {
        this.booling = booling;
    }

    public long getExTime() {
        return exTime;
    }

    public void setExTime(long exTime) {
        this.exTime = exTime;
    }

    private double r;
    private LocalTime birthTime;
    private boolean booling=false;
    private long exTime;



}
